### ISSUE TITLE
DeEntitize fails with KeyNotFoundException 

### ISSUE THREAD

**Issue by Bor1ss** [zzzprojects/html-agility-pack#27]
Hi!
we have some cases when during DeEntitize KeyNotFoundException occurs.

at HtmlEntity.DeEntitize  line 646: 
`object o = _entityValue[entity.ToString()];`
fails if value is not presented in Dictionary

I think the fix is to use TryGetValue

```
if (!_entityValue.TryGetValue(entity.ToString(), out code))
{
    // nope
    sb.Append("&" + entity + ";");
}
```

**Comment by JonathanMagnan** [comment]
Hello @Bor1ss ,

Thank you for reporting. We will look at this issue this weekend.

Best Regards,

Jonathan

**Comment by JonathanMagnan** [comment]
Hello @Bor1ss ,

Your fix has been included in the stable version v1.5.0

Thank again for your contribution.

Best Regards,

Jonathan
